api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/AuthServiceClient.java

```diff
@@ -262,5 +262,18 @@ public Mono<ResponseEntity<TokenResponseDTO>> validateToken(String jwtToken) {
                 .toEntity(TokenResponseDTO.class);
     }
 
+    public Mono<UserResponseDTO> updateUsersRoles(String userId, RolesChangeRequestDTO rolesChangeRequestDTO, String jwToken) {
+        return webClientBuilder.build()
+                .patch()
+                .uri(authServiceUrl + "/users/{userId}", userId)
+                .bodyValue(rolesChangeRequestDTO)
+                .cookie("Bearer", jwToken)
+                .accept(MediaType.APPLICATION_JSON)
+                .retrieve()
+                .onStatus(HttpStatusCode::is4xxClientError, n -> rethrower.rethrow(n,
+                        x -> new GenericHttpException(x.get("message").toString(), (HttpStatus) n.statusCode())))
```

Comment:
My code

Reply:
It's not ALL your code, I already implemented some stuff before and you helped me fix the lines in the implementation that caused problems.

Reply:
When "we" worked on it I changed all of it

Reply:
And all your code was gone

Reply:
No, you mean only my code for the service impl was "gone". Then the rest was already done, just a few changes to adapt to the changes you made. Just admit it, you want me to fail.